### ISSUE TITLE
is there existing a bug? 

### ISSUE THREAD

**Issue by zhl5842** [facebookresearch/llama#218]
https://github.com/facebookresearch/llama/blob/57b0eb62de0636e75af471e49e2f1862d908d9d8/llama/generation.py#L42
logits = self.model.forward(tokens[:, prev_pos:cur_pos], prev_pos) 

to

logits = self.model.forward(tokens[:,:cur_pos], 0)

**Comment by zhl5842** [comment]
https://github.com/facebookresearch/llama/issues/151 done
